.github/workflows/ci-pullrequest-main.yml

```diff
@@ -79,12 +79,44 @@ jobs:
   PackageDeploy:
     runs-on: ubuntu-latest
     needs: BuildTest
+    permissions:
+      contents: read
+      packages: write
+      id-token: write
 
     steps:
     - uses: actions/checkout@v3
     
     - name: Docker Setup BuildX
       uses: docker/setup-buildx-action@v2
+
+    - name: Set image tag
+      run: |
+        echo "DOCKER_TAG_SHA=ghcr.io/buerokratt/mock-classifier:${GITHUB_SHA}" >> $GITHUB_ENV
```

Comment:
Could we set these environment variables under the env section like?
PackageDeploy:
    runs-on: ubuntu-latest
    .....
    env:
      DOCKER_TAG_SHA: ghcr.io/buerokratt/mock-classifier:${GITHUB_SHA}
      DOCKER_TAG_LATEST: ghcr.io/buerokratt/mock-classifier:latest

Reply:
Not all variables can be set at env section at the begging. The problem is that some variables don't have value what are used to create other env values at this stage. For example `$GITHUB_SHA` gets its value after the repo is cloned thus making hard to create tag for docker image with the latest commit SHA.